### ISSUE TITLE
Unable to connect to device on setup

### ISSUE THREAD

**Issue by BoHome88** [humbertogontijo/homeassistant-roborock#390]
Version of vacuum: S7
I have been unable to get this integration to connect to the vacuum on setup. 
I’m currently on version 1.06
![IMG_0688](https://github.com/humbertogontijo/homeassistant-roborock/assets/117252014/d3f3f9fb-7bc7-4519-a3b0-1d4eb7985f6b)

Logger: homeassistant.config_entries
Source: config_entries.py:1247 
First occurred: 7:58:17 PM (1 occurrences) 
Last logged: 7:58:17 PM

Config entry 'xxxxxxxxxx@gmail.com' for roborock integration not ready yet: There are no devices that can currently be reached.; Retrying in background

**Comment by Heisenberg2980** [comment]
Similar issue here, my roborock S7 was fine up until 12 hours ago when it became unavailable in Home Assistant. Restarting the integration or Home Assistant has not fixed the issue

**Comment by prometheanfire** [comment]
can you access https://usaccount.roborock.com/login ? It seems like they are starting to block people (I'm still blocked 48 hours since I noticed).

**Comment by humbertogontijo** [comment]
Can you try with 1.0.8?

**Comment by BoHome88** [comment]
> Can you try with 1.0.8?

I can’t update to 1.08 as it requires HA core update to 2023.6.1 which causes problems with my xiaomi devices. But I think it maybe because I do not have a US roborock account.

**Comment by BoHome88** [comment]
> can you access https://usaccount.roborock.com/login ? It seems like they are starting to block people (I'm still blocked 48 hours since I noticed).

Yes I can login. But the address does not load as I’m outside of the US.

**Comment by BoHome88** [comment]
Can the dev please confirm this integration works for account holders outside of the states?

**Comment by BoHome88** [comment]
I’m here to report that integration still doesn’t connect to device after updating to 1.08…

**Comment by CH1PSET1** [comment]
I have a similar problem. Often the robot loses connection and recovers only after restarting the integration.

Logger: roborock.cloud_api
Source: custom_components/roborock/camera.py:221
Integration: Roborock ([documentation](https://github.com/humbertogontijo/homeassistant-roborock), [issues](https://github.com/humbertogontijo/homeassistant-roborock/issues))
First occurred: 23 июня 2023 г. в 21:11:28 (507 occurrences)
Last logged: 11:40:15

id=68766 Timeout after 4 seconds
id=73304 Timeout after 4 seconds
id=69527 Timeout after 4 seconds
id=99124 Timeout after 4 seconds
id=77446 Timeout after 4 seconds


Logger: roborock.local_api
Source: custom_components/roborock/coordinator.py:78
Integration: Roborock ([documentation](https://github.com/humbertogontijo/homeassistant-roborock), [issues](https://github.com/humbertogontijo/homeassistant-roborock/issues))
First occurred: 23 июня 2023 г. в 22:04:11 (9 occurrences)
Last logged: 09:12:02

Failed connecting to 192.168.4.13:
Failed connecting to 192.168.4.13: [Errno 113] Connect call failed ('192.168.4.13', 58867)
Failed connecting to 192.168.4.13: [Errno 111] Connect call failed ('192.168.4.13', 58867)

**Comment by Heisenberg2980** [comment]
> I have a similar problem. Often the robot loses connection and recovers only after restarting the integration.
> 
> Logger: roborock.cloud_api Source: custom_components/roborock/camera.py:221 Integration: Roborock ([documentation](https://github.com/humbertogontijo/homeassistant-roborock), [issues](https://github.com/humbertogontijo/homeassistant-roborock/issues)) First occurred: 23 июня 2023 г. в 21:11:28 (507 occurrences) Last logged: 11:40:15
> 
> id=68766 Timeout after 4 seconds id=73304 Timeout after 4 seconds id=69527 Timeout after 4 seconds id=99124 Timeout after 4 seconds id=77446 Timeout after 4 seconds
> 
> Logger: roborock.local_api Source: custom_components/roborock/coordinator.py:78 Integration: Roborock ([documentation](https://github.com/humbertogontijo/homeassistant-roborock), [issues](https://github.com/humbertogontijo/homeassistant-roborock/issues)) First occurred: 23 июня 2023 г. в 22:04:11 (9 occurrences) Last logged: 09:12:02
> 
> Failed connecting to 192.168.4.13: Failed connecting to 192.168.4.13: [Errno 113] Connect call failed ('192.168.4.13', 58867) Failed connecting to 192.168.4.13: [Errno 111] Connect call failed ('192.168.4.13', 58867)

@CH1PSET1 that was already happening before which was not a really big deal as you can create an automation to reload the integration when the roborock has been offline for 5 or 10 minutes. The problem now is that for some of us the roborock stays offline even after restarting the integration or even Home Assistant, but at least for me the Roborock app is working.

Does anybody know if Roborock is blocking 3rd party integrations as it seems for some users?

**Comment by Heisenberg2980** [comment]
Some good news everyone, after upgrading to 1.08 my roborock was still offline in Home Assistant, so I decided to remove the integration and added it again and it worked, my roborock is back online. 

Can everyone having this issue remove the Roborock integration and add it again to see if that solves issue?

**Comment by BoHome88** [comment]
Have you got a US account?

**Comment by Heisenberg2980** [comment]
No, I have a UK account

**Comment by CH1PSET1** [comment]
Tell me, how do you see the status of Roborock (online or offline)?

**Comment by BoHome88** [comment]
> Tell me, how do you see the status of Roborock (online or offline)?

The roborock is online on the app. I can control the vacuum through the app. The app even gives a notification to say that I’m logged on another device when I log in through the integration, but the integration can’t connected to the device. I’ve tried deleting and reinstalling the integration. Still doesn’t help.

**Comment by CH1PSET1** [comment]
In my application, Roborock seems to be always available. I understand there is a problem with integration? That somewhere in it is determined the status of Roborock online or offline.

**Comment by Heisenberg2980** [comment]
@BoHome88 sorry to hear that, I was hopping deleting/reinstalling the integration would work for everyone, do you have a US account? can anybody else confirm if reinstalling the integration works for them and what country is their account so we can try to understand if the issue is related to the location?

**Comment by BoHome88** [comment]
> @BoHome88 sorry to hear that, I was hopping deleting/reinstalling the integration would work for everyone, do you have a US account? can anybody else confirm if reinstalling the integration works for them and what country is their account so we can try to understand if the issue is related to the location?

I have a New Zealand account. I have been having trouble with device connection failure since day 1. Setup has never been successful for me…

**Comment by Heisenberg2980** [comment]
It would be good if someone else with a New Zealand account could confirm if they are having the same issue. Also for testing you could create another account in another country using a different email and add the roborock to it to see if that solves your problem

**Comment by BoHome88** [comment]
> It would be good if someone else with a New Zealand account could confirm if they are having the same issue. Also for testing you could create another account in another country using a different email and add the roborock to it to see if that solves your problem

Will I lose the map created under my current account?

**Comment by CH1PSET1** [comment]
> that was already happening before which was not a really big deal as you can create an automation to reload the integration when the roborock has been offline for 5 or 10 minutes.

What does it mean to go offline. How to define this mode? What is the condition to create automation for restarting integration? What lets me know that the vacuum cleaner has gone offline?

**Comment by humbertogontijo** [comment]
Anyone having this problem have the robot reachable from HA instance?
If not try setting option to use cloud instead of local connection and reloading the integration

**Comment by Xyaren** [comment]
> Anyone having this problem have the robot reachable from HA instance? If not try setting option to use cloud instead of local connection and reloading the integration

@humbertogontijo I am running my HA in a docker container, so the network discovery does not work.
The device would be reachable via its IP.
Is there a way to use cloud get the device information and try connecting to the local IP but not rely on the discovery part ?
